### PR TITLE
fixed previous bugs & add ShowMedicationProfile screen with it's functionality

### PR DESCRIPTION
**Pull request by hassan-prog**

### CODE REVIEW

File: grad_login/lib/infrastructure/medicine/interactions_service.dart [hunk 1]
```diff
@@ -23,7 +24,16 @@ class InteractionsService {
         "Authorization": "JWT $token",
       },
     );
-    final responseData = json.decode(response.body);
+    if (response.statusCode == 500) {
+      responseData['error'] =
```
Comment (0xSherlokMo) [inline grad_login/lib/infrastructure/medicine/interactions_service.dart]: You should not mutate the response data at any time, can you split this into another variable? or another object?

File: grad_login/lib/infrastructure/medicine/interactions_service.dart [hunk 2]
```diff
@@ -23,7 +24,16 @@ class InteractionsService {
         "Authorization": "JWT $token",
       },
     );
-    final responseData = json.decode(response.body);
+    if (response.statusCode == 500) {
```
Comment (0xSherlokMo) [inline grad_login/lib/infrastructure/medicine/interactions_service.dart:30]: you can simplify this into breaking early.
```Dart
if(response.statusCode == 500){
   // your logic
   return responseData;
}
responseData = json.decode(response.body);
if (responseData.isEmpty) {
   // your logic
}

return responseData
```
in this way, you're avoiding nested if chain.

**Comment by hassan-prog** [comment]
optimized the code according to your comments, thanks!

**Review by 0xSherlokMo — APPROVED** [review]
LGTM, merge whenever you're ready.

**Pull request merged**